node/libs/crypto/src/bn254/testonly.rs

```diff
@@ -0,0 +1,36 @@
+//! Random key generation, intended for use in testing
+
+use super::{AggregateSignature, PublicKey, SecretKey, Signature};
+use rand::{distributions::Standard, prelude::Distribution, Rng};
+
+/// Generates a random SecretKey. This is meant for testing purposes.
+impl Distribution<SecretKey> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> SecretKey {
+        let rand = ark_bn254::Fr::new(rng.gen());
+        SecretKey(rand)
+    }
+}
+
+/// Generates a random PublicKey. This is meant for testing purposes.
+impl Distribution<PublicKey> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> PublicKey {
+        rng.gen::<SecretKey>().public()
```

Comment:
It's simpler to just generate a random G2 point.

Reply:
Fixed.